nixos/modules/services/web-apps/fresh-rss.nix

```diff
@@ -0,0 +1,219 @@
+{ config, pkgs, lib, ... }:
+
+let
+
+  inherit (lib) mkDefault mkEnableOption mkForce mkIf mkMerge mkOption;
+  inherit (lib) literalExample optional optionalString types;
+
+  cfg = config.services.fresh-rss;
+  fpm = config.services.phpfpm.pools.fresh-rss;
+
+  user = "freshrss";
+  group = config.services.httpd.group;
+  stateDir = "/var/lib/fresh-rss";
+
+in
+{
+  # interface
+  options = {
+    services.fresh-rss = {
+
+      enable = mkEnableOption "FreshRSS";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.fresh-rss;
+        description = "Which FreshRSS package to use.";
+      };
+
+      initialPassword = mkOption {
+        type = types.str;
+        example = "correcthorsebatterystaple";
+        description = ''
+          Specifies the initial password for the admin, i.e. the password assigned if the user does not already exist.
+          The password specified here is world-readable in the Nix store, so it should be changed promptly.
+        '';
+      };
+
+      database = {
+        host = mkOption {
+          type = types.str;
+          default = "localhost";
+          description = "Database host address.";
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 3306;
+          description = "Database host port.";
+        };
+
+        name = mkOption {
+          type = types.str;
+          default = "freshrss";
+          description = "Database name.";
+        };
+
+        user = mkOption {
+          type = types.str;
+          default = "freshrss";
+          description = "Database user.";
+        };
+
+        passwordFile = mkOption {
+          type = types.nullOr types.path;
+          default = null;
+          example = "/run/keys/freshrss-dbpassword";
+          description = ''
+            A file containing the password corresponding to
+            <option>database.user</option>.
+          '';
+        };
+
+        createLocally = mkOption {
+          type = types.bool;
+          default = true;
+          description = ''
+            Create the database and database user locally.
```

Comment:
To me this description doesn't make completely clear what this option is about. Together with the assertion below I assume that this is used to create a mysql db on the same machine with socket authentication?

Reply:
Thanks for this feedback. You are correct. Any suggestions?

Reply:
```nix
localDB = mkOption {
  type = types.bool;
  default = true;
  description = ''
    Create and use a database on the same machine. This allows using authetication to the db without any credentials.
    This currently only applies if database type "mysql" is selected.
  '';
};
```